### DISCUSSION TITLE
Make DatePicker and RangePicker Input Fields Read-Only

### DISCUSSION THREAD

**Discussion by Hashim1995** [heroui-inc/heroui#3255] [Help]
I am using the DatePicker and DateRangePicker components from NextUI, and I would like to enhance the user experience by disabling manual typing in the date fields. Users should only be able to select dates by opening the picker dialog. This prevents any format mistakes and ensures the input is controlled exclusively by the picker.

Expected Behavior
Both DatePicker and DateRangePicker should have a prop, perhaps readOnlyInput, that when set to true, makes the input fields read-only. Users can still activate the picker dialog to select dates but cannot type or edit dates directly in the input field.

Current Behavior
Currently, users can type directly into the input fields of both DatePicker and DateRangePicker. There is no official prop to make these input fields read-only

**Comment by marsidorowicz** [comment]
great idea, wondering how to do this without a fix or feat, but nothing works
